File: servicepractice/src/main/java/com/example/servicepractice/MainActivity.java
package com.example.servicepractice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button startButton, stopButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        startButton = findViewById(R.id.btn_start_service);
        stopButton = findViewById(R.id.btn_stop_service);

        startButton.setOnClickListener(this);
        stopButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (view == startButton){
            Toast.makeText(this,"Service Started", Toast.LENGTH_LONG).show();
            startService(new Intent(this, ServiceActivity.class));
        } else if (view == stopButton){
            Toast.makeText(this,"Service Stopped", Toast.LENGTH_LONG).show();
            stopService(new Intent(this, ServiceActivity.class));
        }
    }
}

File: app/src/main/java/com/fwhyn/android/MainActivity.kt
package com.fwhyn.android

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.fwhyn.android.ServicesPractice.ServiceMainActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // initiate button to start Service Activity
        val btnOpenServiceActivity = findViewById<Button>(R.id.btn_open_service_activity)

        // intent to Service Activity
        btnOpenServiceActivity.setOnClickListener {
            val intent = Intent(this, ServiceMainActivity::class.java)
            startActivity(intent)
        }
    }
}